### PR TITLE
Rdsimaps 3346 sample map link copy link

### PR DESCRIPTION
**Pull request by acarrera42**
Exposes "Copy Link" button on Sample Maps.

### CODE REVIEW

File: lib/assets/javascripts/cartodb/public_common/sample_settings_view.js [hunk 1]
```diff
@@ -79,6 +81,39 @@ module.exports = cdb.core.View.extend({
...
   },
 
+  _copyLink: function() {
+    // Display Bloomberg terminal link
+    // Link is of form: {MAPS SAMPLE:<ID>}
+    var bbg_link = '{MAPS SAMPLE:' + this.vizId + '}';
+
+     var link = this.currentUser.viewUrl() + "/api/v1/sample_link/" + this.vizId;
+     var html = `
+        <div class="u-inner">
```
Comment (vishalya) [inline lib/assets/javascripts/cartodb/public_common/sample_settings_view.js]: Please separate out the html code into an ejs and then render, if this code is going to be extended in future.

**Review by vishalya — CHANGES_REQUESTED** [review]

**Comment by acarrera42** [comment]
Updated to use template.  Please re-review.  Thanks.

**Comment by vishalya** [comment]
Looks good.

File: lib/assets/javascripts/cartodb/public_common/sample_settings_view.js [hunk 2]
```diff
@@ -84,23 +84,38 @@ module.exports = cdb.core.View.extend({
...
+
+    // Create instance object.
+    var dialog = new CopyLinkDialog();
+
+    // To render & show initially (only to be called once):
+    dialog.appendToBody();
+
+    // Copy to clipboard
+    var clipboardLink = document.querySelector('.js-bbg-link');
+    var range = document.createRange();
```
Comment (vishalya) [inline lib/assets/javascripts/cartodb/public_common/sample_settings_view.js]: js-bbg-link selector sounds very generic, what will happen if there are more than 1 on the page? Will that condition happen?

Comment (acarrera42) [reply lib/assets/javascripts/cartodb/public_common/sample_settings_view.js]: If there's more than one I think it becomes ambiguous what will be copied.  Although I don't think anyone else will use it in this popup I can change it to be js-bbg-sample-link

**Pull request merged**